api/projects/permissions.py

```diff
@@ -59,6 +66,30 @@ def has_object_permission(self, request, view, obj):
         return False
 
 
+class MasterAPIKeyProjectPermissions(BasePermission):
+    def has_permission(self, request: HttpRequest, view: str) -> bool:
+        master_api_key = getattr(request, "master_api_key", None)
+
+        if not master_api_key:
+            return False
+
+        if view.action == "create":
+            organisation = int(request.data.get("organisation"))
+            return organisation == master_api_key.organisation_id
+
+        if view.action in ("list", "permissions"):
+            return True
+
+        # move on to object specific permissions
+        return view.detail
+
+    def has_object_permission(
+        self, request: HttpRequest, view: str, obj: Project
+    ) -> bool:
+        master_api_key = request.master_api_key
+        return master_api_key.organisation_id == obj.organisation.id
```

Comment:
Sorry to be picky, but we should use `organisation_id` on both sides of this equality check

Reply:
yep, for sure.